chain/consensus/iface.go

```diff
@@ -18,12 +18,35 @@ import (
 )
 
 type Consensus interface {
-	ValidateBlock(ctx context.Context, b *types.FullBlock) (err error)
+	// ValidateBlockHeader is called by peers when they receive a new block through the network.
+	//
+	// This is a fast sanity-check validation performed by the PubSub protocol before delivering
+	// it to the syncer. It checks that the block has the right format and it performs
+	// other consensus-specific light verifications like ensuring that the block is signed by
+	// a valid miner, or that it includes all the data required for a full verification.
```

Comment:
By "through the network" you mean through gossipsub, right?
If I understood your PR description correctly, nodes that participate in the BFT protocol itself would not use this. That makes sense, since for a BFT-style protocol like the Mir-based one we have, this only seems meaningful (possibly) a part of some state synchronization sub-protocol (or for "learner" nodes, if we introduce such a thing within the protocol).

For this to be useful, we need to define what exactly a block header must look like, how it will be validated, by whom, and what information is necessary to validate the block (e.g. the membership?).